### PR TITLE
Switch client to use plugin SDK client

### LINKED ISSUE #83: Make use of Grafana HTTP client
Refactor backend to make use of the HTTP client from `grafana-plugin-sdk-go` rather than the client from the `http` package. This allows us to use middlewares that enable forwarding of auth headers.

### PR DESCRIPTION
**Pull request by aangelisc**
- Only set token header if ForwardHTTPHeaders is false
- Remove unneeded TLSConfig code
- Add test

I've done some rudimentary testing to validate I can still create a datasource and query but the TLS settings still need to be verified.

Fixes #83 

Part of #85

### CODE REVIEW

**Review by asimpson — APPROVED** [review]
Removing 3x the code you're adding?! :trophy: for you!

> I've done some rudimentary testing to validate I can still create a datasource and query but the TLS settings still need to be verified.

Did you figure out a way to verify this? Once we can verify it I'm good with merging.

File: pkg/humio/client.go [hunk 1]
```diff
@@ -22,15 +20,8 @@ type Client struct {
...
 type Config struct {
-	Address            *url.URL
-	Token              string
-	InsecureSkipVerify bool
-	TlsClientAuth      bool
-	TlsAuthWithCACert  bool
-	TlsCACert          string
-	TlsClientCert      string
-	TlsClientKey       string
-	TlsServerName      string
```
Comment (bossinc) [inline pkg/humio/client.go:33]: Do we need to backport these fields? Do old data source still work?

File: pkg/humio/client.go [hunk 2]
```diff
@@ -109,65 +100,22 @@ func (c *Client) GraphQLQuery(query interface{}, variables map[string]interface{
...
-		Transport: &HttpHeaderTransport{
-			rt:      rt,
-			headers: headers,
-		},
-	}, nil
-}
-
-// getTLSConfig returns tlsConfig from settings
-// logic reused from https://github.com/grafana/grafana/blob/615c153b3a2e4d80cff263e67424af6edb992211/pkg/models/datasource_cache.go#L211
-func getTLSConfig(config Config) (*tls.Config, error) {
```
Comment (bossinc) [inline pkg/humio/client.go:149]: How is TLS being handled now?

Comment (aangelisc) [reply pkg/humio/client.go:149]: My understanding is that TLS is handled by the default HTTP client [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/30591d3456466e8e7004ef4cbb5427e0a618066f/backend/httpclient/http_client.go#L99) once a new client is created [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/30591d3456466e8e7004ef4cbb5427e0a618066f/backend/httpclient/http_client.go#L20).

File: pkg/humio/client.go (hunk 1 above)
Comment (aangelisc) [reply pkg/humio/client.go:33]: They should still work as in the frontend the settings were still controlled via the `DefaultHTTPSettings` component so the default http client in the backend should pick them up fine.

**Review by bossinc — APPROVED** [review]

**Pull request merged**